packages/runtime/container-runtime/src/summaryBlobProtocol.ts

```diff
@@ -0,0 +1,128 @@
+ * - HEADER CONTENT :  HEADER CONTENT LENGTH bytes
+ */
+export class BlobHeader {
+    public static readonly ENCODING = "ascii";
+    public static readonly HEADER_PREFIX = Buffer.from("f4e72832a5bd47d7a2f35ed47ed94a3c", "hex");
+    constructor(private readonly _fields: { [key: string]: string; }) { }
+    public getValue(key: string): string {
+        return this._fields[key];
+    }
+
+    /**
+     * This method returns the length of the binary representation of this header object.
+     * @returns Returns the length of the binary representation of this header object.
+     */
+    public headerLength() {
+        return this.toBinary().byteLength;
+    }
+
+    /**
+     * This method converts this header object to binary array.
+     * @returns this header object as binary array.
+     */
+    public toBinary(): Buffer {
+        const contentStr = JSON.stringify(this._fields);
+        const contentBinary: Buffer = Buffer.from(contentStr, BlobHeader.ENCODING);
+        const contentBinaryLength = contentBinary.byteLength;
+        const contentBinaryLengthBuf = new ArrayBuffer(4);
+        const view = new DataView(contentBinaryLengthBuf);
+        view.setUint32(0, contentBinaryLength, false);
+        return Buffer.concat([BlobHeader.HEADER_PREFIX, Buffer.from(contentBinaryLengthBuf), contentBinary]);
+    }
+
+    /**
+     * This method converts the binary array to header object, if not present, undefined is returned.
+     * @returns header object
+     */
+    public static fromBinary(buffer: Buffer): BlobHeader | undefined {
+        const possibleHeader = buffer.slice(0, BlobHeader.HEADER_PREFIX.byteLength);
+        if (!possibleHeader.equals(BlobHeader.HEADER_PREFIX)) {
+            return undefined;
+        }
+        const arrayBuffer = toArrayBuffer(buffer);
+        const view = new DataView(arrayBuffer);
+        const contentBinaryLength = view.getUint32(BlobHeader.HEADER_PREFIX.byteLength, false);
+        const contentPos = BlobHeader.HEADER_PREFIX.byteLength + 4;
+        const contentBinary = arrayBuffer.slice(contentPos, contentPos + contentBinaryLength);
+        const contentStr = Buffer.from(contentBinary).toString(BlobHeader.ENCODING);
+        return new BlobHeader(JSON.parse(contentStr));
+    }
+
+    /**
+     * This method returns the rest of the message cutting of the complete header from the given byte array.
+     * @param buffer - The byte array
+     * @returns The rest of the message cutting of the complete header from the given byte array.
+     */
+    public static skipHeader(buffer: Buffer): Buffer {
+        const header = this.fromBinary(buffer);
+        if (!this.fromBinary(buffer)) {
+            return buffer;
+        } else {
+            return buffer.slice(header?.headerLength());
+        }
+    }
+}
+
+function toArrayBuffer(buf) {
+    const ab = new ArrayBuffer(buf.length);
+    const view = new Uint8Array(ab);
```

Comment:
According to documentation, constructor copies content of array into typed array. Which suggests we are wasting CPU resources here, I believe. I believe IsoBuffer has conversions tools that do this kind of conversion.

Reply:
Implemented in f16b576ffe91c8c6593bfd311588970a03632a28